cli/2024-10-cli-udt.md

```diff
@@ -0,0 +1,150 @@
+|-------------|-------------|---------|
+|Resource provider | Refers to a group of resource types under a common namespace. Resource providers are also entities that implement the group of resource types | Mycompany.Messaging or Applications.Core |
+|Resource type | A service or technology that can be modeled as a resource | PostgreSQL or AWS S3 or Internal messaging service |
+|Resource | An instance of a resource type | mydb or mybucket |
+
+#### User Experience
+
+1. Deb uses the rad CLI to scaffold a `yaml` template definition for the internal messaging service `Plaid`
+
+    Option 1:
+
+    ```bash
+    rad resource-provider init Mycompany.Messaging --schema-format yaml
+    ```
+        
+    This command will create a folder with the name `Mycompany.Messaging`. A sample `yaml` file is created in the application folder. If we enable typespec, a sample `tsp` will be created and all the required dependencies will be created.
+
+    Pros : Depicts the action of initializing the project with directory structure and dependencies. TypeSpec has similar `init` command to set up the project and would be consistent with that when we support typespec.
+
+    Option 2:
+
+    ```bash
+    rad resource-type new Mycompany.Messaging --schema-format yaml
+    ```
+    
+    Deb updates the schema definition with the required fields and properties for the messaging service Plaid. The authoring experience for the resource type will be covered in a separate document.
+
+2. Deb registers the resource type with Radius
+
+    Option 1 
+
+    ```bash
+    rad resource-provider create Mycompany.Messaging
+    ```
+    Pros : Follows the CRUDL convention and is more consistent with the other commands in Radius
+
+    Option 2
+
+    ```bash
+    rad resource-provider register Mycompany.Messaging
+    ```
+    Pros : Consistent with Recipe template registration to environment
+    Cons : Does not follow the CRUDL convention
+
+3. Deb creates the resource instance for the messaging service Plaid
+
+    ```bash
+    rad resource create Mycompany.Messaging/Plaid -f myplaid.json
+    ```
+    The resource `myplaid` of type `Mycompany.Messaging/Plaid`is created and available for use.
+
+4. Deb lists the resource providers and resource types available in Radius
+
+    ```bash
+    rad resource-provider list
+    ```
+    The list of resource providers in Radius are displayed.
+
+    ```bash
+    rad resource-type list --resource-provider Mycompany.Messaging
+    ```
+    The list of resource types under the resource provider `Mycompany.Messaging` are displayed.
+
+5. Deb makes changes to the resource type definition of `Plaid` and adds to the resource provider
+
+    Option 1:
+
+    ```bash
+    rad resource-provider create Mycompany.Messaging
+    ```
+    Pros : Same command can be used to update the resource provider with the new or updated resource type definitions
+
+    Option 2 :
+
+    ```bash
+    rad resource-type create plaid.yaml 
```

Comment:
It doesn't seem like we have a consistent way of referring to an outside file in the CLI. Should it be -f? @? Or maybe a flag isn't necessary at all? I just think we should be consistent. We use -f earlier.

Reply:
yeah there should be a `-f` here.